src/t_set.c

```diff
@@ -59,6 +59,7 @@ robj *setTypeCreate(sds value, size_t size_hint) {
 /* Check if the existing set should be converted to another encoding based off the
  * the size hint. */
 void setTypeMaybeConvert(robj *set, size_t size_hint) {
+    /* Because this is an existing set, there will be at least one element, so we use >= */
     if ((set->encoding == OBJ_ENCODING_LISTPACK && size_hint >= server.set_max_listpack_entries)
         || (set->encoding == OBJ_ENCODING_INTSET && size_hint >= server.set_max_intset_entries))
```

Comment:
```suggestion
    if ((set->encoding == OBJ_ENCODING_LISTPACK && size_hint > server.set_max_listpack_entries)
        || (set->encoding == OBJ_ENCODING_INTSET && size_hint > server.set_max_intset_entries))
```
Yeah, I think these are also all wrong.